### PR TITLE
Add evaluation

### LINKED ISSUE #1: Clickthrough rate

### PR DESCRIPTION
**Pull request by KristofferHaakonsen**
Closes: #1 
Add evaluation at the end of the pipeline.
We now vary the number of recommended articles we evaluate, so that we get more data. We also do evaluations for all the data. This makes it easier to discuss the results and the prioritization.

We are currently calculating Recall, ARHR, and CTR.
results

### CODE REVIEW

File: evaluation.py [hunk 1]
```diff
@@ -1,21 +1,27 @@
-def evaluate(pred, actual, k, num_requests=1, mse = 0.5):
+def evaluate(pred, actual, k, num_requests):
     """
-    Evaluate recommendations according to recall@k and ARHR@k
+    Evaluate recommendations according to recall@k, ARHR@k, and
```
Comment (Pernilleww) [inline evaluation.py]: and ctr?

Comment (KristofferHaakonsen) [reply evaluation.py]: Haha, and ctr 👍 
Nice catch!

**Comment by jonasrodningen** [comment]
ready to merge?

**Comment by KristofferHaakonsen** [comment]
> ready to merge?

Ja, venter på godkjenning :)
Men hva tenker dere om ARHR?

**Comment by jonasrodningen** [comment]
added a major overhaul, fixing a HUGE bug and edited recall + CTR metrics

File: evaluation.py [hunk 2]
```diff
@@ -1,21 +1,47 @@
...
+            if pred in test_set_positives_list:
+                # the prediction/suggestion is a true positive (tp)
+                tp_global += 1.
+                arhr += 1. / float(ind + 1.)
+
+
+    '''
+    print("num_actually " ,num_actual)
+    print("num_users : " , num_users)
+    print("num_requests : " , num_requests)
```
Comment (Pernilleww) [inline evaluation.py]: Fjerne denne kommentaren?

Comment (KristofferHaakonsen) [reply evaluation.py]: Ja, det er nok lurt :)

**Pull request merged**